sdk/iot/provisioning/src/az_iot_provisioning_client.c

```diff
@@ -10,3 +10,175 @@
+  {
+    AZ_RETURN_IF_FAILED(az_span_append_uint8(mqtt_user_name, '&', &mqtt_user_name));
+    AZ_RETURN_IF_FAILED(az_span_append(mqtt_user_name, *user_agent, &mqtt_user_name));
+  }
+
+  *out_mqtt_user_name = mqtt_user_name;
+  return AZ_OK;
+}
+
+// <registration_id>
+AZ_NODISCARD az_result az_iot_provisioning_client_id_get(
+    az_iot_provisioning_client const* client,
+    az_span mqtt_client_id,
+    az_span* out_mqtt_client_id)
+{
+  AZ_PRECONDITION_NOT_NULL(client);
+  AZ_PRECONDITION_VALID_SPAN(mqtt_client_id, 0, false);
+  AZ_PRECONDITION_NOT_NULL(out_mqtt_client_id);
+
+  AZ_RETURN_IF_FAILED(
+      az_span_copy(mqtt_client_id, client->_internal.registration_id, &mqtt_client_id));
+
+  *out_mqtt_client_id = mqtt_client_id;
+  return AZ_OK;
+}
+
+// $dps/registrations/res/#
+AZ_NODISCARD az_result az_iot_provisioning_client_register_subscribe_topic_filter_get(
+    az_iot_provisioning_client const* client,
+    az_span mqtt_topic_filter,
+    az_span* out_mqtt_topic_filter)
+{
+  UNUSED(client);
+
+  AZ_PRECONDITION_NOT_NULL(client);
+  AZ_PRECONDITION_VALID_SPAN(mqtt_topic_filter, 0, false);
+  AZ_PRECONDITION_NOT_NULL(out_mqtt_topic_filter);
+
+  AZ_RETURN_IF_FAILED(az_span_copy(mqtt_topic_filter, str_dps, &mqtt_topic_filter));
+  AZ_RETURN_IF_FAILED(az_span_append(mqtt_topic_filter, str_registrations, &mqtt_topic_filter));
+  AZ_RETURN_IF_FAILED(az_span_append(mqtt_topic_filter, str_res, &mqtt_topic_filter));
+  AZ_RETURN_IF_FAILED(az_span_append_uint8(mqtt_topic_filter, '#', &mqtt_topic_filter));
+
+  *out_mqtt_topic_filter = mqtt_topic_filter;
+  return AZ_OK;
+}
+
+// topic: $dps/registrations/res/202/?$rid=1&retry-after=3
+// payload:
+// {"operationId":"4.d0a671905ea5b2c8.42d78160-4c78-479e-8be7-61d5e55dac0d","status":"assigning"}
+AZ_NODISCARD az_result az_iot_provisioning_client_received_topic_payload_parse(
+    az_iot_provisioning_client const* client,
+    az_span received_topic,
+    az_span received_payload,
+    az_iot_provisioning_client_register_response* out_response)
+{
+  UNUSED(client);
+  UNUSED(received_topic);
+  UNUSED(received_payload);
+  UNUSED(out_response);
+
+  return AZ_ERROR_NOT_IMPLEMENTED;
+}
+
+// $dps/registrations/PUT/iotdps-register/?$rid={%s}
+AZ_NODISCARD az_result az_iot_provisioning_client_register_publish_topic_get(
+    az_iot_provisioning_client const* client,
+    az_span mqtt_topic,
+    az_span* out_mqtt_topic)
+{
+  UNUSED(client);
+
+  AZ_PRECONDITION_NOT_NULL(client);
+  AZ_PRECONDITION_VALID_SPAN(mqtt_topic, 0, false);
+  AZ_PRECONDITION_NOT_NULL(out_mqtt_topic);
+
+  AZ_RETURN_IF_FAILED(az_span_copy(mqtt_topic, str_dps, &mqtt_topic));
+  AZ_RETURN_IF_FAILED(az_span_append(mqtt_topic, str_registrations, &mqtt_topic));
+  AZ_RETURN_IF_FAILED(az_span_append(mqtt_topic, str_put_iotdps_register, &mqtt_topic));
+
+  *out_mqtt_topic = mqtt_topic;
+  return AZ_OK;
+}
+
+// Topic: $dps/registrations/GET/iotdps-get-operationstatus/?$rid={%s}&operationId=%s
```

Comment:
nit: should it be
```suggestion
// Topic: $dps/registrations/GET/iotdps-get-operationstatus/?$rid={%s}&operationId={%s}
```

Reply:
I don't think so: why would it have {}? I think that's just a marker that will be replaced by the string itself :)
I may be wrong. Do we have it in other places? Can you try w/o? (Saving 2 chars over the wire is good.)

Reply:
I wasn't sure if this was you demonstrating where the strings for those values would go or not. You had it for `$rid={%s}` and wasn't sure if that was actually sent on the wire or if that's your placeholder for what goes there.

Reply:
More to this:
1. OperationID is not containing {} in the json payload.
2. RID has {rid} in the IoT Hub and DPS documentation but works just fine w/o them.

Reply:
> You had it for $rid={%s} and wasn't sure if that was actually sent on the wire or if that's your placeholder for what goes there.

You're right: I've fixed all of them to remove the {} because it's confusing.